core/src/main/scala/org/apache/spark/scheduler/DecommissionInfo.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.scheduler
+
+/**
+ * Provides more detail about a decommissioning event
+ * @param message human readable reason for why the decommissioning is happening
+ * @param workerAlsoDecommissioned whether the worker is being decommissioned too.
+ *                                 Used to know if the shuffle data might be lost too.
+ */
+private[spark]
+case class DecommissionInfo(message: String, workerAlsoDecommissioned: Boolean)
```

Comment:
Maybe, `workerDecommissioned`?

Reply:
We probably need a base class for decom or just extends `ExecutorLossReason`... Then, we could use the base class or `ExecutorLossReason` in `TaskScheduler.executorDecommission`. Thereform, `TaskScheduler.executorDecommission` could support other cluster managers as well.

Reply:
DecommissionInfo is just an `info` struct. It's not the loss reason itself, so it shouldn't overload ExecutorLossReason. 

I didn't get the part about TaskScheduler.executorDecommission supporting other cluster managers: TaskScheduler is about task scheduling not coarse grained executor scheduling. I think it should be totally unaware of whether the decommission signal came via Yarn or K8s or whatever other cluster management strategy we have.

Reply:
But now, it looks like `DecommissionInfo` is specific to Standalone because of the terminology “worker”. 

I agree that `TaskScheduler` should be unaware of where the decom signal comes. That's why I asked for the base decom class. However, it's still ok if `DecommissionInfo` could be applied to other clusters as well.

Reply:
Fair enough. I can see why the "Worker" nomenclature is confusing. Although `ExecutorProcessLost` has a `workerLost` field that can (in theory) be set by any scheduler backend. I will rename/reword to make this independance clearer.